### DISCUSSION TITLE
About Trajectory CSV File

### DISCUSSION THREAD

**Discussion by imamim** [USC-ACTLab/crazyswarm#618] [Q&A] [answered]
Hello have a good day.

I have couple of question for trajectory's .csv file and `uav_trajectory.py`. 

Firstly, in this `ros_ws/src/crazyswarm/scripts/figure8.csv` we have x,y,z,yaw value, but for both (and other all csv) we have 7 value of each element.I'm a bit confused here, can you explain what the values ​​in this csv are and why there are 7 of each? Does this code only contain 7th order polynomials as in the `Online Single-Piece Polynomials` section described in the paper?

Also, does the eval function in uav_trajectory.py get the speed value by applying the Mellinger control?

And lastly when i use `cmd_FullState`, is this command doing any controller for reach the target velocity ?
Thanks for answer :)

**Comment by whoenig** [comment, marked as answer]
The values are the coefficients of a polynomial. You can find the mathematical details by looking at the plotting script at https://github.com/whoenig/uav_trajectories/blob/main/scripts/plot_trajectory.py.

The velocity/speed is computed by using the differential flatness property, not by assuming a particular controller.

cmd_fullState just sends a desired state to the quadcopter. It depends on the selected controller what is done with that state. If you use the Mellinger controller, the CF will try to track all components (i.e., position, velocity, yaw angle, and angular velocity).

**Reply by imamim** [reply]
I understand, thanks for information.
